packages/client/hmi-client/src/workflow/ops/model-comparison/model-comparison-operation.ts

```diff
@@ -1,6 +1,15 @@
 import { Operation, WorkflowOperationTypes } from '@/types/workflow';
 
-export interface ModelComparisonOperationState {}
+export interface NotebookHistory {
+	code: string;
+	timestamp: number;
+}
+
+export interface ModelComparisonOperationState {
+	notebookHistory: NotebookHistory[];
+	hasCodeRun: boolean;
+	structuralComparisons: string[];
+}
```

Comment:
Can you give us some ideas and/or examples of what `notebookHistory` and `structualComparisons` are? 

We want to be careful of not jamming too much information into a workflow because it is saved as a single document, so subsequent saving and retrieval can become quite prohibitive.

Reply:
`structuralComparisons` hold the images that compare the models. 

As for `notebookHistory` I am following a similar pattern to `model-edit` and `model-config` notebooks where the code is saved within `codeHistory`. I was thinking to keep things consistent. There isn't an output planned for this component though so I am also thinking to just save a normal `code` state.

Reply:
It may be worthwhile to take an hour to see if we can maybe save the image(s) to Redis instead of into the workflow...I am not sure what support we have in place so if it is too annoying to do I am okay with saving into workflow - but it will for sure slow things down a bit.

E.g. thinking something along this lines:
```
image = fetch_image_from_redis(imageId)
if image === null:
  image = genrate_image(code_history_or_whateever)
  imageId = cache_image(image)
```

Reply:
Is that you [volunteering](https://github.com/DARPA-ASKEM/terarium/blob/main/packages/client/hmi-client/src/workflow/ops/model-edit/tera-model-edit.vue#L317)